src/pages/workspaces/workspace/useDeleteWorkspaceState.ts

```diff
@@ -35,83 +53,150 @@ export interface DeleteWorkspaceHookArgs {
         setWorkspaceBucketUsageInBytes(usageInBytes)
-      } else {
-        const currentWorkspaceAppList = await Ajax(signal).Apps.listAppsV2(workspaceInfo.workspaceId)
-
-        // temporary hack to prevent orphaning resources on Azure:
-        // change each app to status: 'disallow' which will cause this modal to think they are undeletable
-        const hackedAppList = _.map(_.set('status', 'disallow'), currentWorkspaceAppList) as DeleteWorkspaceModalLeoApp[]
-        setApps(hackedAppList)
-
-        // Also temporarily disable delete if there are any controlled resources besides the expected workspace storage container.
-        const controlledResources = await Ajax(signal).WorkspaceManagerResources.controlledResources(workspaceInfo.workspaceId)
-        setControlledResourcesExist(controlledResources.resources.length > 1)
       }
     })
     load()
+
+    clearTimeout(checkAzureResourcesTimeout.current)
   })
 
   const hasApps = () => {
-    return deletableApps !== undefined && nonDeletableApps !== undefined &&
-        (!_.isEmpty(deletableApps) ||
-            !_.isEmpty(nonDeletableApps))
+    return (workspaceResources !== null && !_.isEmpty(workspaceResources.apps))
+  }
+
+  const hasRuntimes = () => {
+    return (workspaceResources !== null && !_.isEmpty(workspaceResources.runtimes))
   }
 
-  const isDeleteDisabledFromResources = (hasApps() && !_.isEmpty(nonDeletableApps)) || controlledResourcesExist
+  const isDeleteDisabledFromResources = workspaceResources !== null && (
+    (hasApps() && !_.isEmpty(workspaceResources.nonDeleteableApps)) || (hasRuntimes() && !_.isEmpty(workspaceResources.nonDeleteableRuntimes))
+  )
 
   const deleteWorkspace = async () => {
+    if (isDeleteDisabledFromResources) {
+      throw new Error('Workspace contains non-deletable resources')
+    }
+
     try {
       setDeleting(true)
-      if (isGoogleWorkspace(hookArgs.workspace)) {
+      if (isGoogleWorkspace(hookArgs.workspace) && workspaceResources) {
         await Promise.all(
-          _.map(async app => await Ajax().Apps.app(app.cloudContext.cloudResource, app.appName).delete(), deletableApps)
+          _.map(async app => await Ajax(signal).Apps.app(app.cloudContext.cloudResource, app.appName).delete(), workspaceResources.deleteableApps)
         )
       }
-      await Ajax().Workspaces.workspace(workspaceInfo.namespace, workspaceInfo.name).delete()
+
+      await Ajax(signal).Workspaces.workspace(workspaceInfo.namespace, workspaceInfo.name).delete()
       hookArgs.onDismiss()
       hookArgs.onSuccess()
     } catch (error) {
-      reportError('Error deleting workspace', error)
       setDeleting(false)
+      await reportError('Error deleting workspace', error)
+    }
+  }
+
+  const checkAzureResources = async () => {
+    console.log('Checking azure resources...') // eslint-disable-line no-console
+    const appsInfo = await fetchWorkspaceResources(hookArgs.workspace)
```

Comment:
If `fetchWorkspaceResources` throws a server error, there is nothing to handle it here (unlike at https://github.com/DataBiosphere/terra-ui/pull/3800/files#diff-218b26c3d13f09c1d0c645847ac2e875e72df6b133ddebea46b34f189a86425cR97, where `withErrorReportingInModal` is being used).

Unhandled server errors cause confusing situations that are difficult to diagnose.